src/payment/strategies/stripev3/stripev3-payment-strategy.ts

```diff
@@ -0,0 +1,129 @@
+
+export default class StripeV3PaymentStrategy implements PaymentStrategy {
+    private stripeJs?: StripeV3Js;
+    private cardElement?: StripeCardElement;
+
+    constructor(
+        private _store: CheckoutStore,
+        private _paymentMethodActionCreator: PaymentMethodActionCreator,
+        private _paymentActionCreator: PaymentActionCreator,
+        private _orderActionCreator: OrderActionCreator,
+        private _stripeScriptLoader: StripeScriptLoader
+    ) {}
+
+    initialize(options: PaymentInitializeOptions): Promise<InternalCheckoutSelectors> {
+        const DOMElement = options.stripev3;
+
+        if (!DOMElement) {
+            throw new InvalidArgumentError('Unable to initialize payment because "options.stripe" argument is not provided.');
+        }
+
+        const paymentMethod = this._store.getState().paymentMethods.getPaymentMethod(options.methodId);
+
+        if (!paymentMethod) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        return this._stripeScriptLoader.load(paymentMethod.initializationData.stripePublishableKey)
+            .then((stripeJs: StripeV3Js) => {
+                this.stripeJs = stripeJs;
+                const elements = this.stripeJs.elements();
+                this.cardElement = elements.create('card', {
+                    style: DOMElement.elementProps,
+                });
+
+                if (!this.cardElement) {
+                    throw new InvalidArgumentError('Unable to initialize payment because "StripeCardElement" argument is not provided.');
+                }
+
+                this.cardElement.mount('#' + DOMElement.containerId);
+
+                return Promise.resolve(this._store.getState());
+            });
+    }
+
+    execute(payload: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { payment, ...order } = payload;
+
+        if (!payment) {
+            throw new PaymentArgumentInvalidError(['payment.paymentData']);
+        }
+
+        return this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(payment.methodId))
+            .then(state => {
+                const paymentMethod = state.paymentMethods.getPaymentMethod(payment.methodId);
+
+                if (!paymentMethod || !paymentMethod.clientToken) {
+                    throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+                }
+
+                return this._getStripeJs().handleCardPayment(
+                    paymentMethod.clientToken, this._getCardElement(), {}
+                ).then((stripeResponse: StripeResponse) => {
+                    if (stripeResponse.error) {
+                        throw new StandardError(stripeResponse.error.message);
+                    } else {
+                        if (!stripeResponse.paymentIntent.id) {
+                            throw new PaymentArgumentInvalidError(['paymentIntent.id']);
```

Comment:
`PaymentArgumentInvalidError` is only appropriate when the SDK user forgets to pass in a required parameter. So it shouldn't be used here as `paymentIntent.id` is not something passed in by the SDK user.

I had a look at the doc, it seems `id` would only be undefined if `error` is defined. So you could probably combine the check with the above. i.e.:

```ts
if (response.error || !response.paymentIntent.id) {
    throw new StandardError(response.error && response.error.message);
}
```